packages/pipeline-editor/src/validation.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// 3 seconds should be ample time.
+// should only protect against any infinite loops.
+const TIMEOUT = 3 * 1000;
```

Comment:
When do you expect to get here? If we are trying to identify infinite loops, should we stop when we find the first occurrence of trying to visit a vertice that has been already processed as that identifies a circular dependency?

Reply:
There should never never be an infinite loop (unless there is a bug my the algorithm) this is just a safety net for that case